Deploy Cachito

athos-ribeiro opened the pull request:
* CLOUDBLD-203

Signed-off-by: Athos Ribeiro <athos@redhat.com>

chmeliik commented on deploy.yaml:
```diff
@@ -317,3 +318,114 @@
       changed_when: true
+
+
+- name: Run cachito containers on openshift
```
Does this play do the equivalent of [setting up dirs for koji PVs](https://github.com/containerbuildsystem/osbs-box/blob/24eb5bed0485ad68d981b8fdde9338490623cc1d/deploy.yaml#L135-L142)? If not, do you know why it works?

athos-ribeiro replied:
Added so we set the proper SELinux flags.
TBH, idk why it worked before, maybe due to the way ansible created the parent directory with the equivalent play for koji? :)

chmeliik commented on deploy.yaml:
```diff
@@ -317,3 +318,114 @@
+      - cachito-rabbitmq
+      - cachito-athens
+  tags:
+    - cachito
```
`cachito` tag should also be added to `Prepare for deployment of openshift applications`

athos-ribeiro commented on cachito-api/Dockerfile:
```diff
@@ -0,0 +1,36 @@
+# Install Cachito from git
+ENV PIP_PREFIX="/usr"
+RUN pip install "$CACHITO_PIP_REF"
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
```
Learned it from the GH linter warning! https://docs.docker.com/engine/reference/builder/#shell

mprahl commented on openshift/templates/cachito-volumes.yaml:
```diff
@@ -0,0 +1,71 @@
+        - ReadWriteOnce
+      resources:
+        requests:
+          storage: 100Gi
```
This seems rather high for a dev environment

athos-ribeiro replied:
We are just not limiting the resources for those volumes. I did it this way to keep it consistent to the other PVs, such as the koji and registry ones

mprahl commented on openshift/templates/cachito-volumes.yaml:
```diff
@@ -0,0 +1,71 @@
+        - ReadWriteMany
+      resources:
+        requests:
+          storage: 100Gi
```
This seems rather high for a dev environment

athos-ribeiro commented on koji-db/sql-init/30-cachito.sql:
```diff
@@ -0,0 +1,4 @@
+CREATE USER cachito WITH PASSWORD 'mypassword';
+CREATE DATABASE cachito;
+GRANT ALL PRIVILEGES ON DATABASE cachito TO koji;
```
It was not, it is just a remaining from when I started working on the playbooks :)
Removed!

athos-ribeiro commented:
Thanks for the reviews :)

I addressed your comments and also included a new commit to use cachito workers cert auth. This means that this now relies on https://github.com/release-engineering/cachito/pull/166. Please use that branch to test this if the PR is not merged.

chmeliik commented on openshift/templates/cachito-api.yaml:
```diff
@@ -0,0 +1,140 @@
+    cachito-api
+
+labels:
+  build: cachito-api
```
nitpick: `app` instead of `build` (otherwise it shows up under `Other resources` in OpenShift console)

chmeliik commented on openshift/templates/cachito-rabbitmq.yaml:
```diff
@@ -0,0 +1,81 @@
+        kind: Service
+        name: cachito-rabbitmq
+      port:
+        targetPort: "15672"
```
This should be `targetPort: "80"`, it targets the service port, not the container port. The OpenShift console warns about this (that's how I noticed), though the route seems to work anyway.

chmeliik commented on openshift/templates/cachito-worker.yaml:
```diff
@@ -0,0 +1,140 @@
+    cachito-worker
+
+labels:
+  build: cachito-worker
```
nitpick: `app` instead of `build` (otherwise it shows up under `Other resources` in OpenShift console)

athos-ribeiro commented:
Thanks, Adam!

I Addressed the inline comments. You are right on the routes. I dropped the  ones for rabbitmq and the cachito worker.

The cachito API one is definitelly useful, but just for debugging purposes. I am not sure if providing the client cert is needed at this point (same applies for the other 2 routes).

The cachito PR has been merged, in case anyone else wants to give this a try :)
